taxi/views.py

```diff
@@ -79,6 +82,28 @@
     success_url = reverse_lazy("taxi:car-list")
 
 
+class AssignDriverView(LoginRequiredMixin, View):
+    def post(self, request, pk, *args, **kwargs):
+        car = get_object_or_404(Car, pk=pk)
+        if request.user in car.drivers.all():
+            messages.info(request, "You are already a driver of this car.")
+        else:
+            car.drivers.add(request.user)
+            messages.success(request, "You have been assigned to this car.")
+        return redirect("taxi:car-detail", pk=pk)
+
+
+class RemoveDriverView(LoginRequiredMixin, View):
+    def post(self, request, pk, *args, **kwargs):
+        car = get_object_or_404(Car, pk=pk)
+        if request.user in car.drivers.all():
+            car.drivers.remove(request.user)
```

Comment:
In `RemoveDriverView`, ensure that only `Driver` instances are being removed from cars. You should verify that `request.user` is an instance of `Driver` before removing them from `car.drivers`.